app/models/user_session.rb

```diff
@@ -70,4 +72,10 @@ def user_intervention_finished?
 
     user_intervention.completed_sessions == user_intervention.sessions.size
   end
+
+  def user_intervention_inside_health_clinic?
+    return false if user_intervention.nil?
+
+    user_intervention.health_clinic_id.present?
+  end
```

Comment:
It's Okay, but what do you think about removing this logic and using `delegate_to` instead?